### PR TITLE
Carousel fixes

### PR DESCRIPTION
**Pull request by patrickhlauke**
Replace touch-action: none with pan-y, remove preventDefault from touch event handling

This makes scrolling and carousels coexist. Some minor drawbacks. that I guess we can live with:

- it doesn't prevent browser gestures in touch-events scenarios (primarily iOS, where swiping from the side will trigger the browser's back/forward functionality, but this is common across these scenarios)
- Firefox on touchscreen-enabled laptops/desktops with high-dpi will throw `pointercancel` and abort a swipe gesture as soon as there's even a minimal amount of vertical movement to the swipe - this seems to be a bug/overeager behavior on the part of FX; testing on a medium-dpi Android, Firefox there seems to behave sanely

Closes https://github.com/twbs/bootstrap/issues/27531

/cc @MartijnCuppens @Johann-S

### CODE REVIEW

**Comment by MartijnCuppens** [comment]
Can you remove the generated css/js?

**Comment by patrickhlauke** [comment]
oh heaven's sake, i thought i had ... yup

**Comment by patrickhlauke** [comment]
the dangers of getting confused while testing locally. removed all the dist rubbish and force-pushed

File: js/src/carousel.js [hunk 1]
```diff
@@ -283,14 +283,11 @@ class Carousel {
       if (this._pointerEvent && (event.originalEvent.pointerType === PointerType.TOUCH || event.originalEvent.pointerType === PointerType.PEN)) {
         this.touchStartX = event.originalEvent.clientX
       } else if (!this._pointerEvent) {
-        event.preventDefault()
         this.touchStartX = event.originalEvent.touches[0].clientX
       }
     }
 
     const move = (event) => {
-      event.preventDefault()
```
Comment (Johann-S) [inline js/src/carousel.js]: I don't understand why do you remove this line ? 🤔
This function `move` is used only for device which do not handle PointerEvents and prevent the "go back" behaviour

Comment (patrickhlauke) [reply js/src/carousel.js]: `preventDefault` on `touchstart` and `touchmove` prevents scrolling (one of the browser's "default actions") in touch-events-only browsers (e.g. iOS Safari). leaving this in would still result in carousels not allowing scrolling on iPhone etc.

**Review by MartijnCuppens — APPROVED** [review]
Tested on several devices, everything seem to work fine. Couldn't test this on a touch desktop device with firefox (this isn't available on BrowserStack either I think).

**Comment by patrickhlauke** [comment]
i tested on my Surface Pro (2017) Win 10/Firefox 63 - this is the only browser that currently fails to react to the swipe, due to the Surface's high dpi/Firefox's overeager `pointercancel`. Planning to file a bug against FX sometime this weekend, after having mentioned it already at last week's TPAC to @smaug----

**Comment by patrickhlauke** [comment]
can't seem to merge this until a code owner reviews...

**Comment by XhmikosR** [comment]
Then wait for it :P

**Review by Johann-S — APPROVED** [review]
LGTM 👍 sorry for the delay

**Comment by patrickhlauke** [comment]
Closing the loop here: Firefox bug https://bugzilla.mozilla.org/show_bug.cgi?id=1502010

**Pull request merged**